src/GovUk.Education.ExploreEducationStatistics.Data.Services.Tests/ResultSubjectMetaServiceTests.cs

```diff
@@ -1828,6 +2202,237 @@ await using (var statisticsDbContext = InMemoryStatisticsDbContext(statisticsDbC
+                    subjectRepository: subjectRepository.Object,
+                    timePeriodService: timePeriodService.Object,
+                    options: options
+                );
+
+                var result = await service.GetSubjectMeta(
+                    releaseId,
+                    query,
+                    observations);
+
+                MockUtils.VerifyAllMocks(
+                    boundaryLevelRepository,
+                    filterItemRepository,
+                    footnoteRepository,
+                    indicatorRepository,
+                    locationRepository,
+                    releaseDataFileRepository,
+                    subjectRepository,
+                    timePeriodService);
+
+                var viewModel = result.AssertRight();
+
+                var locationViewModels = viewModel.LocationsHierarchical;
+
+                // Result has Region and Local Authority levels
+                Assert.Equal(2, locationViewModels.Count);
+
+                var regions = locationViewModels["region"];
+
+                Assert.Equal(3, regions.Count);
+                var regionOption1 = regions[0];
+                var regionOption2 = regions[1];
+                var regionOption3 = regions[2];
+
+                Assert.Equal(_northEast.Name, regionOption1.Label);
+                Assert.Equal(_northEast.Code, regionOption1.Value);
+
+                Assert.Equal(_northWest.Name, regionOption2.Label);
+                Assert.Equal(_northWest.Code, regionOption2.Value);
+
+                Assert.Equal(_eastMidlands.Name, regionOption3.Label);
+                Assert.Equal(_eastMidlands.Code, regionOption3.Value);
+
+                // Expect a hierarchy of Region-LA within the Local Authority level
+                var localAuthorities = locationViewModels["localAuthority"];
+                Assert.Equal(3, localAuthorities.Count);
+
+                var laOption1 = localAuthorities[0];
+                Assert.NotNull(laOption1);
+                Assert.Equal(_northEast.Name, laOption1.Label);
+                Assert.Equal(_northEast.Code, laOption1.Value);
+                Assert.Equal("region", laOption1.Level);
+                Assert.Single(laOption1.Options!);
+
+                var laOption2 = localAuthorities[1];
+                Assert.NotNull(laOption2);
+                Assert.Equal(_northWest.Name, laOption2.Label);
+                Assert.Equal(_northWest.Code, laOption2.Value);
+                Assert.Equal("region", laOption2.Level);
+                Assert.Single(laOption2.Options!);
+
+                var laOption3 = localAuthorities[2];
+                Assert.NotNull(laOption3);
+                Assert.Equal(_eastMidlands.Name, laOption3.Label);
+                Assert.Equal(_eastMidlands.Code, laOption3.Value);
+                Assert.Equal("region", laOption3.Level);
+                Assert.Single(laOption2.Options!);
```

Comment:
There's a mistake here

Reply:
Great spot, fixed! :smile:

Yeh there's a few things that I needed to do. The location de-duping stuff on the frontend was the most pain tbh. It'll be good when we can remove it all in EES-2783.

Reply:
That will be great when we can remove it all! 🤞 

I can't see this fix yet btw. 🙂